pkg/controller/controller_instance.go

```diff
@@ -2021,3 +1832,191 @@ func shouldStartOrphanMitigation(statusCode int) bool {
+			return true, nil
+		}
+	}
+
+	return false, nil
+}
+
+// cloneServiceInstance clones the given ServiceInstance. This is so that the
+// reconciler may modify the cloned resource without accidentally mutating the
+// cache.
+func cloneServiceInstance(instance *v1beta1.ServiceInstance) (*v1beta1.ServiceInstance, error) {
+	clone, err := api.Scheme.DeepCopy(instance)
+	if err != nil {
+		return nil, err
+	}
+	return clone.(*v1beta1.ServiceInstance), nil
+}
+
+// updateServiceInstanceReconciliationError handles the logging and updating of
+// a ServiceInstance that hit a retryable error during reconciliation.
+func (c *controller) updateServiceInstanceReconciliationError(instance *v1beta1.ServiceInstance, reason, message string) error {
+	pcb := pretty.NewContextBuilder(pretty.ServiceInstance, instance.Namespace, instance.Name)
+
+	instance, err := cloneServiceInstance(instance)
+	if err != nil {
+		return fmt.Errorf(pcb.Messagef("error cloning: %v", err))
+	}
+
+	glog.Info(pcb.Message(message))
+	c.recorder.Event(instance, corev1.EventTypeWarning, reason, message)
+
+	setServiceInstanceCondition(
+		instance,
+		v1beta1.ServiceInstanceConditionReady,
+		v1beta1.ConditionFalse,
+		reason,
+		message,
+	)
+	if _, err := c.updateServiceInstanceStatus(instance); err != nil {
+		return err
+	}
+
+	// The result of this function should be directly returned from the
+	// reconciler, so it is necessary to return an error to tell the worker
+	// to retry reconciling the resource.
+	//
+	// TODO(mkibbe): Change worker semantics so that retries do not rely on
+	// an error return value.
+	return fmt.Errorf(message)
+}
+
+// updateServiceInstanceDeprovisionAsyncResponse handles the logging and
+// updating of a ServiceInstance that received an asynchronous response from
+// the broker when requesting a deprovision.
+func (c *controller) updateServiceInstanceDeprovisionAsyncResponse(instance *v1beta1.ServiceInstance, response *osb.DeprovisionResponse) error {
+	pcb := pretty.NewContextBuilder(pretty.ServiceInstance, instance.Namespace, instance.Name)
+
+	instance, err := cloneServiceInstance(instance)
+	if err != nil {
+		return fmt.Errorf(pcb.Messagef("error cloning: %v", err))
+	}
+
+	glog.Info(pcb.Message(asyncDeprovisioningMessage))
+	c.recorder.Event(instance, corev1.EventTypeNormal, asyncDeprovisioningReason, asyncDeprovisioningMessage)
+
+	if response.OperationKey != nil && *response.OperationKey != "" {
+		key := string(*response.OperationKey)
+		instance.Status.LastOperation = &key
+	}
+
+	instance.Status.AsyncOpInProgress = true
+
+	setServiceInstanceCondition(
+		instance,
+		v1beta1.ServiceInstanceConditionReady,
+		v1beta1.ConditionFalse,
+		asyncDeprovisioningReason,
+		asyncDeprovisioningMessage,
+	)
+	if _, err := c.updateServiceInstanceStatus(instance); err != nil {
+		return err
+	}
+
+	return c.beginPollingServiceInstance(instance)
+}
+
+// updateServiceInstanceReconciliationError handles the logging and updating of
```

Comment:
Fix comment so that it is for the `updateServiceInstanceDeprovisionSuccess` function.